erpnext/projects/doctype/task/task.json

```diff
@@ -381,14 +387,20 @@
    "fieldname": "duration",
    "fieldtype": "Int",
    "label": "Duration (Days)"
+  },
+  {
+   "depends_on": "eval: doc.status == \"Completed\"",
+   "fieldname": "completed_on",
+   "fieldtype": "Date",
+   "label": "Completed On"
```

Comment:
Let's keep the "Completed On" field mandatory on changing status to Completed. Also, add a validation to check "Completed On" does not exceed the current date.